src/core/algorithms/md/hymd/lattice_traverser.h

```diff
@@ -12,7 +12,14 @@ namespace algos::hymd {
 
 class LatticeTraverser {
 private:
+    struct LatticeStatistics {
+        std::size_t all_mds_num = 0;
+        std::size_t invalidated_mds_num = 0;
+        static constexpr double kRatioBound = 0.01;
+    };
+
     Recommendations recommendations_;
+    LatticeStatistics lattice_statistics_;
```

Comment:
These statistics should only be collected for the current level, not for the whole lattice traversal